src/components/donation-flow/steps/Amount.tsx

```diff
@@ -94,19 +105,19 @@ export default function Amount() {
               .concat({ label: t('first-step.other'), value: 'other' }) || []
           }
         />
-        <Collapse unmountOnExit in={amount.value === 'other'} timeout="auto">
+        <Collapse unmountOnExit in={value === 'other'} timeout="auto">
           <Grid
             item
             xs={12}
             sm={6}
             style={
-              !mobile
+              !mobile && Number(prices?.length) % 2 === 0
                 ? {
                     float: 'right',
                     marginTop: -50,
                     width: '49%',
                   }
-                : { marginTop: theme.spacing(2) }
+                : { marginTop: theme.spacing(2), width: mobile ? '100%' : '49%' }
```

Comment:
Can we use `vw` instead of `%` to match the viewport size?

Reply:
This is not matching the viewport size, but making sure the component is 50% because the Grid is with 2 columns on desktop and 1 on mobile. It is essentially a workaround, because we can't put this field inside of the Grid.

<img width="368" alt="image" src="https://user-images.githubusercontent.com/61479393/216430440-762ee3eb-ce29-45c6-93bd-557808d50176.png">
<img width="610" alt="image" src="https://user-images.githubusercontent.com/61479393/216430477-1bbce11c-9686-4b6d-96b6-65b2f444d12f.png">